com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Utility/Logic/AndNode.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using static UnityEditor.ShaderGraph.Registry.Types.GraphType;
+
+namespace com.unity.shadergraph.defs
+{
+
+    internal class AndNode : IStandardNode
+    {
+        public static FunctionDescriptor FunctionDescriptor => new(
+            1,     // Version
+            "And", // Name
+            "Out = A && B;",
+            new ParameterDescriptor("A", TYPE.Bool, Usage.In),
+            new ParameterDescriptor("B", TYPE.Bool, Usage.In),
```

Comment:
Does the TYPE.Bool also accepts vectors? In the original node, the input port can accept vectors as well. Or do we want to change that to bool/float only?

Reply:
When you say "accept" - this is actually two different behaviors.  The first one is to take the data type as is and use it, and the second is to automatically convert it and then use the converted version.  In the old system, you can connect a float to a bool input port, but it gets automatically converted to a boolean (indicated by the wire changing from light blue to purple.) And that's what this will do as well.  If I specify that the input port is a boolean, I will be able to connect a float to it, but it'll be converted.  If I were to specify this input port as a Vector, I could connect a float to it, and it would attempt to use that float value (without type casting) with the && - which can't be done.